ui/stop_details/OBAGenericStopViewController.m

```diff
@@ -475,7 +475,12 @@ - (BOOL)sectionNeedsExplanatoryFooterMessage:(NSInteger)section {
 }
 
 - (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section {
-    return kSectionHeaderHeight;
+    if ([_filteredArrivals count] != [_allArrivals count] && section == 1) {
```

Comment:
please use properties and dot notation whenever possible, e.g. `self.filteredArrivals.count`

Also, instead of checking the raw section index, please check `-sectionTypeForSection:` to make sure that it's the expected section type.